nova/objects/instance_group.py

```diff
@@ -34,14 +34,23 @@
 LOG = logging.getLogger(__name__)
 
 
-def _instance_group_get_query(context, id_field=None, id=None):
+def _instance_group_get_query(context, id_field=None, id=None,
+                              project_id=None,
+                              limit=None,
+                              offset=None):
     query = context.session.query(api_models.InstanceGroup).\
         options(orm.joinedload('_policies')).\
         options(orm.joinedload('_members'))
     if not context.is_admin:
         query = query.filter_by(project_id=context.project_id)
     if id and id_field:
         query = query.filter(id_field == id)
+    if project_id is not None:
```

Comment:
I would move that under
```
if not context.is_admin:
    query = query.filter_by(project_id=context.project_id)
elif project_id is not None:
 ...
```
The filtering by project id only makes sense as admin

Reply:
fixed